modules/core/src/smithy4s/capability/MonadThrowLike.scala

```diff
@@ -22,6 +22,12 @@ trait MonadThrowLike[F[_]] extends Zipper[F] {
   def flatMap[A, B](fa: F[A])(f: A => F[B]): F[B]
   def raiseError[A](e: Throwable): F[A]
   def handleErrorWith[A](fa: F[A])(f: Throwable => F[A]): F[A]
+  def onError[A](fa: F[A])(f: PartialFunction[Throwable, F[Unit]]): F[A] = {
```

Comment:
note: I pushed an override of this for Cats interop specifically - in case there's a more optimal implementation in the cats instance